exercises/practice/dominoes/dominoes.spec.js

```diff
@@ -0,0 +1,104 @@
+import { compileWat, WasmRunner } from "@exercism/wasm-lib";
+
+let wasmModule;
+let currentInstance;
+
+function canChain(input = "") {
+  const inputBufferOffset = 256;
+  const inputBufferCapacity = 256;
+
+  const inputLengthEncoded = new TextEncoder().encode(input).length;
+  if (inputLengthEncoded > inputBufferCapacity) {
+    throw new Error(
+      `String is too large for buffer of size ${inputBufferCapacity} bytes`
+    );
+  }
+
+  currentInstance.set_mem_as_utf8(inputBufferOffset, inputLengthEncoded, input);
+
+  // Pass offset and length to WebAssembly function
+  return currentInstance.exports.canChain(
+    inputBufferOffset,
+    inputLengthEncoded
+  );
+}
+
+beforeAll(async () => {
+  try {
+    const watPath = new URL("./dominoes.wat", import.meta.url);
+    const { buffer } = await compileWat(watPath);
+    wasmModule = await WebAssembly.compile(buffer);
+  } catch (err) {
+    console.log(`Error compiling *.wat: \n${err}`);
+    process.exit(1);
+  }
+});
+
+describe("canChain()", () => {
+  beforeEach(async () => {
+    currentInstance = null;
+    if (!wasmModule) {
+      return Promise.reject();
+    }
+    try {
+      currentInstance = await new WasmRunner(wasmModule);
+      return Promise.resolve();
+    } catch (err) {
+      console.log(`Error instantiating WebAssembly module: ${err}`);
+      return Promise.reject();
+    }
+  });
+
+  test("empty input = empty output", () => {
+    expect(canChain("")).toBe(1);
+  });
+
+  xtest("singleton input = singleton output", () => {
+    expect(canChain("\x11")).toBe(1);
```

Comment:
It would be better to have some automatic conversion so is is easier to read and maintain these tests.